File: src/test/java/com/vicen/films/application/queries/GetFilmQueryTest.java
package com.vicen.films.application.queries;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

class GetFilmQueryTest {

    @Test
    @DisplayName("It instances the query from all its values.")
    void intanceCompleteQuery() {
        UUID uuid = UUID.randomUUID();
        GetFilmQuery query = new GetFilmQuery(uuid);
        assertEquals(uuid, query.getFilmId());
    }

    @Test
    @DisplayName("It instances the query from null values.")
    void instanceQueryFromNullValues() {
        GetFilmQuery query = new GetFilmQuery(null);
        assertNull(query.getFilmId());
    }

}

File: src/test/kotlin/com/vicen/films/application/queries/GetFilmQueryTest.kt
package com.vicen.films.application.queries

import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

import java.util.UUID

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNull

internal class GetFilmQueryTest {

    @Test
    @DisplayName("It instances the query from all its values.")
    fun instanceCompleteQuery() {
        val uuid = UUID.randomUUID()
        val (filmId) = GetFilmQuery(uuid)
        assertEquals(uuid, filmId)
    }

}
